src/main/java/wolox/training/models/User.java

```diff
@@ -52,6 +61,17 @@
     )
     private List<Book> books = new ArrayList<>();
 
+    public User() {
+        // Added to use with JPA;
+    }
+
+    public User(Long id, String username, String name, LocalDate birthDate) {
+        this.id = id;
```

Comment:
I used setters because they contains the **Preconditions** validators so if some field doesn't fulfill the requirements I think the object mustn't be created.

Reply:
Ok,  but is bad practice using methods that can be overridden, if you need yo validate something do it out side the constructor or use annotations. For example look [this](https://docs.oracle.com/javaee/7/tutorial/bean-validation003.htm) and [this](https://www.sitepoint.com/using-java-bean-validation-method-parameters-return-values/)

Reply:
I gonna annotate them